src/OpenTelemetry.AutoInstrumentation/ContinuousProfiler/ContinuousProfilerProcessor.cs

```diff
@@ -0,0 +1,80 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+#if NET6_0_OR_GREATER
+
+using System.Globalization;
+using OpenTelemetry.AutoInstrumentation.Logging;
+
+namespace OpenTelemetry.AutoInstrumentation.ContinuousProfiler;
+
+internal static class ContinuousProfilerProcessor
+{
+    public const string BackgroundThreadName = "OpenTelemetry Continuous Profiler Thread";
+
+    private static readonly IOtelLogger Logger = OtelLogging.GetLogger();
+
+    public static void Activity_CurrentChanged(object? sender, System.Diagnostics.ActivityChangedEventArgs e)
+    {
+        var currentActivity = e.Current;
+        var managedThreadId = Environment.CurrentManagedThreadId;
+
+        if (currentActivity != null)
+        {
+            var hexTraceId = currentActivity.TraceId.ToHexString();
+
+            if (ulong.TryParse(hexTraceId.AsSpan(0, 16), NumberStyles.HexNumber, CultureInfo.InvariantCulture, out var traceIdHigh) &&
+                ulong.TryParse(hexTraceId.AsSpan(16), NumberStyles.HexNumber, CultureInfo.InvariantCulture, out var traceIdLow) &&
+                ulong.TryParse(currentActivity.SpanId.ToHexString(), NumberStyles.HexNumber, CultureInfo.InvariantCulture, out var spanId))
+            {
+                NativeMethods.ContinuousProfilerSetNativeContext(traceIdHigh, traceIdLow, spanId, managedThreadId);
+                return;
+            }
+        }
+
+        NativeMethods.ContinuousProfilerSetNativeContext(0, 0, 0, managedThreadId);
+    }
+
+    public static void Initialize(bool threadSamplingEnabled, bool allocationSamplingEnabled, TimeSpan exportInterval, object continuousProfilerExporter)
+    {
+        Logger.Debug("Initializing Continuous Profiler export thread.");
+
+        var continuousProfilerExporterType = continuousProfilerExporter.GetType();
+        var exportThreadSamplesMethod = continuousProfilerExporterType.GetMethod("ExportThreadSamples");
+
+        if (exportThreadSamplesMethod == null)
+        {
+            Logger.Warning("Exporter does not have ExportThreadSamples method. Continuous Profiler initialization failed.");
+            return;
+        }
+
+        var exportAllocationSamplesMethod = continuousProfilerExporterType.GetMethod("ExportAllocationSamples");
+        if (exportAllocationSamplesMethod == null)
+        {
+            Logger.Warning("Exporter does not have ExportAllocationSamples method. Continuous Profiler initialization failed.");
+            return;
+        }
+
+        var thread = new Thread(() =>
```

Comment:
Is there a way to dispose this thread?

Reply:
1. `Thread` is not implementing `IDisposable`. 
2. This thread is marked as background. [It will be killed by the runtime when all foreground tasks will be finished](https://learn.microsoft.com/en-us/dotnet/standard/threading/foreground-and-background-threads).
3. There is not need to care if all data are correctly _exported_. By the nature, we are sampling. In the worst case the process of handling one buffer will be not handled correctly. IMO it is fine.

Technically, we could pass cancellation token and call it from 'Instrumentation.OnExit'. But it might extend killing process time. I would keep it as is, but if you thing otherwise, we can improve in follow up issue. What do you thint?

Reply:
Same here, leave a TODO we could investigate the need for it later.

Reply:
Done: 93a5e853fb88cd4d3f053057281b3ea25eee0531 #3216